indexing-service/src/main/java/org/apache/druid/indexing/common/task/batch/parallel/PartitionLocation.java

```diff
@@ -31,6 +33,11 @@
  * This class represents the intermediary data server where the partition of {@link #interval} and
  * {@link #getBucketId()} is stored.
  */
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
```

Comment:
Should the default implementation be set to GenericPartitionLocation for backward compatibility during rolling upgrade?